cmd/rocsp-tool/main.go

```diff
@@ -118,31 +157,365 @@ func main2() error {
 		return fmt.Errorf("'issuers' section of config JSON is required.")
 	}
 	clk := cmd.Clock()
-	client, err := rocsp_config.MakeClient(&c.ROCSPTool.Redis, clk)
+	redisClient, err := rocsp_config.MakeClient(&c.ROCSPTool.Redis, clk)
 	if err != nil {
 		return fmt.Errorf("making client: %w", err)
 	}
 
-	for _, respFile := range flag.Args() {
-		err := storeResponse(respFile, issuers, client, clk)
+	var db *sql.DB
+	var ocspGenerator capb.OCSPGeneratorClient
+	if c.ROCSPTool.LoadFromDB != nil {
+		lfd := c.ROCSPTool.LoadFromDB
+		db, err = configureDb(&lfd.DB)
+		if err != nil {
+			return fmt.Errorf("connecting to DB: %w", err)
+		}
+
+		ocspGenerator, err = configureOCSPGenerator(lfd.GRPCTLS,
+			c.ROCSPTool.LoadFromDB.OCSPGeneratorService, clk, metrics.NoopRegisterer)
+		if err != nil {
+			return fmt.Errorf("configuring gRPC to CA: %w", err)
+		}
+		setDefault(&lfd.Speed.RowsPerSecond, 2000)
+		setDefault(&lfd.Speed.ParallelSigns, 100)
+	}
+
+	if len(flag.Args()) < 1 {
+		helpExit()
+	}
+
+	ctx := context.Background()
+	cl := client{
+		issuers:       issuers,
+		redis:         redisClient,
+		db:            db,
+		ocspGenerator: ocspGenerator,
+		clk:           clk,
+	}
+	switch flag.Args()[0] {
+	case "store":
+		for _, respFile := range flag.Args() {
+			respBytes, err := ioutil.ReadFile(respFile)
+			if err != nil {
+				return fmt.Errorf("reading response file %q: %w", respFile, err)
+			}
+			err = cl.storeResponse(ctx, respBytes, nil)
+			if err != nil {
+				return err
+			}
+		}
+	case "load-from-db":
+		if c.ROCSPTool.LoadFromDB == nil {
+			return fmt.Errorf("config field LoadFromDB was missing")
+		}
```

Comment:
Maybe also worth erroring out if `len(flag.Args()) > 1`, since then it looks like someone passed some ocsp response filenames on the command line and we're about to just ignore them all.